eos/eos.hpp

```diff
@@ -448,6 +448,137 @@ template <typename T> class RelativisticEOS {
   Real cl_, cl2_;
 };
 
+template <typename T> class EAPRampEOS {
+  
+  // calculate alpha given a particular pressure
+  PORTABLE_FUNCTION
+  Real alpha_p(const Real P) const {
+    if (P < Pe_) {
+      return alpha0_;
+    } else if (P > Pc_) {
+      return 1.0;
+    } else {
+      return alpha0_ + (P - Pe_) / (Pc_ - Pe_) * (1.0 - alpha0_);
+    }
+  }
+  
+  template<typename F, typename ... ARGS>
+  PORTABLE_FUNCTION
+  auto alpha_impl(F func, const Real rho, ARGS... args) const {
+    constexpr const Real a_eps {1.e-12};
+    Real a {0.0};
+    Real a_old {alpha0_};
+    Real diff {0.0};
+    Real P {0.0};
+    for (int i {0}; i < 10; ++i) {
+      P = *func(a_old*rho, args...) / a_old;
+      a = this->alpha_p(P);
+      diff = fabs(a - a_old) / fabs(a + a_old) * 2.0;
+      if (diff < a_eps) break;
+      a_old = a;
+    }
+    return std::tie(a, P);
```

Comment:
I think we should move away from from fixed point iteration quickly. It's slow and not trustworthy. I'd rather use the root finding utilities we already developed.

Reply:
@Yurlungur will we always have access to the solver machinery? Perhaps using the machinery if spiner is available and use this fixed point (or some other easy to implement solve) when spiner isn't available via some `#ifdef`s. I should note that my initial implementation had duplicated solver stuff all over the code, so I was pretty pleased with myself for this one, though this has yet to be fully compiled, I should probably instantiate a concrete object at some point to make sure this actually works lol.

Reply:
@dholladay00 The solver machinery is in `singularity-eos/utils` always. It's based in `singularity-eos`, so I think it's safe to assume it's always available. Will fixed point always work?

Reply:
We are now using this, one notable downside is that using the solver I'd need to create a custom functor to obtain the pressure at the correct alpha. At the moment I'm just using a lambda and an extra pressure call.